rotkehlchen/chain/optimism/modules/velodrome/decoder.py

```diff
@@ -36,11 +63,219 @@ def __init__(
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address == ZERO_ADDRESS
+            ):
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} after depositing in velodrome pool {tx_log.address}'  # noqa: E501
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
+                event.counterparty = CPT_VELODROME
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_remove_liquidity_events(
+            self,
+            tx_log: 'EvmTxReceiptLog',
+            decoded_events: list['EvmEvent'],
+    ) -> DecodingOutput:
+        """Decodes events that remove liquidity from a velodrome v1 or v2 pool"""
+        for event in decoded_events:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address in self.pools
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Return {event.balance.amount} {crypto_asset.symbol}'
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address in self.pools
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from velodrome pool {tx_log.address}'  # noqa: E501
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
+                event.counterparty = CPT_VELODROME
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_swap(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes events that swap eth or tokens in a velodrome v1 or v2 pool"""
+        spend_event, receive_event = None, None
+        for event in context.decoded_events:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                (event.address in (ROUTER_V2, ROUTER_V1) or event.address in self.pools)
```

Comment:
wouldn't it make sense to add something like `self.protocol_addresses = {ROUTER_V2, ROUTER_v1}.union(self.pools)` in the constructor? And then use this everywhere with a single `in` check here?